recipes/wpewebkit/0010-Workaround-for-Android-emulator-samplerExternalOES-p.patch

```diff
@@ -0,0 +1,31 @@
+From 7f7283277042e993c1a847e59d45bb08c7a6250e Mon Sep 17 00:00:00 2001
+From: Jani Hautakangas <jani@igalia.com>
+Date: Sun, 22 May 2022 23:03:03 +0300
+Subject: [PATCH] Workaround for Android emulator samplerExternalOES precision
+ issue
+
+For some unknown reason glShaderSource fails with 0x505 GL_OUT_OF_MEMORY
+on Android Emulator if precision is not set right after SamplerExternalOESType
+definition. Seems like emulator OpenGL ES implementation doesn't define any
+precision for samplerExternalOES. Adding precision hint for samplerExternalOES
+across all platforms is harmless because OpenGL ES implementation is not
+required to use hint information.
+---
+ .../platform/graphics/texmap/TextureMapperShaderProgram.cpp      | 1 +
+ 1 file changed, 1 insertion(+)
+
+diff --git a/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp b/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
+index 299f0a9e..83b325c0 100644
+--- a/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
++++ b/Source/WebCore/platform/graphics/texmap/TextureMapperShaderProgram.cpp
+@@ -167,6 +167,7 @@ static const char* vertexTemplateCommon =
+     GLSL_DIRECTIVE(ifdef ENABLE_TextureExternalOES) \
+         GLSL_DIRECTIVE(extension GL_OES_EGL_image_external : require) \
+         GLSL_DIRECTIVE(define SamplerExternalOESType samplerExternalOES) \
++        STRINGIFY(precision mediump samplerExternalOES;\n) \
```

Comment:
Nicer! This may even have some chances of getting accepted upstream.

Searching around a bit more I have found [how a quirk was added in Skia](https://skia-review.googlesource.com/c/skia/+/235279), and it basically emits this line in shader sources for the Android emulator. If the folks in upstream WebKit won't accept the patch as-is, an option could be to do something a bit more surgical like Skia.